specification/source/cxf.rst

```diff
@@ -0,0 +1,290 @@
+
+All the ElementaryBlock within the standard have been 
+defined and included in ``CXF-Core.jsonld``. However, CXF 
+representation of elementary blocks does not contain 
+the implementation details of the blocks. 
+
+Generating CXF from an instance of a CDL class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A CXF representation of a sequence will be able to be 
+generated from a instance of a CDL class. When there 
+are instances of a CDL class within a Modelica or 
+another CDL class, if the instance has the CDL 
+annotation ``__cdl(export=true)``, the corresponding CDL 
+class shall be translated to CXF. If there is no 
+``export`` annotation attached to the instance, the 
+default value would be false and no CXF will be 
+generated for the class of that instance. 
+
+Source of CXF translation
+^^^^^^^^^^^^^^^^^^^^^^^^^
+The source and version of the tool that generated CXF
+representation shall be optionally included using the 
+properties ``translationSoftware`` and 
+``translationSoftwareVersion`` respectively to the CXF
+block representation. 
+
+Representing Instances in CXF
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Instances of a CDL class shall contain the entire 
+package path of the CDL class, the octothorpe character
+(``#``) followed by the name of the instance in CXF. 
+An instance ("child") of an instance ("parent") shall 
+be referenced by the parent instance’s CXF 
+representation, followed by a period character (``.``)
+and then the child instance’s name. When such a child 
+instance is present in a CDL sequence, the CXF 
+representation of the parent instance shall contain a 
+``hasInstance`` property to the child instance. 
+
+[Example of a CDL instance representation in CXF
+
+CDL:
+  
+.. code-block:: modelica
+
+    within ExamplePackage;
+    block ExampleSeq
+    CDL.Reals.MultiplyByParameter gain(k = 100000 
+        "My gain";
+    end ExampleSeq;
+
+CXF reference to ``gain`` instance: ``ExamplePackage.ExampleSeq#gain``
+CXF reference to ``gain.k`` instance: ``ExamplePackage.ExampleSeq#gain.k``
+CXF property linking ``gain`` and ``gain.k``: ``ExamplePackage.ExampleSeq#gain S231:hasInstance ExamplePackage.ExampleSeq#gain.k .``
+
+Handling Vectors and Expressions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A CXF translation of a CDL sequence shall optionally 
+include certain configuration options that specify 
+how the translation will handle:
+
+* Vectors: A tool that generates CXF translations 
+  from CDL shall optionally include a configuration
+  indicating whether or not to flatten or 
+  preserve the vector references. By default, 
+  vector references in CDL should be preserved 
+  in CXF. If vector references should be flattened, 
+  an index appearing within square 
+  brackets (``[`` and ``]``) in CDL shall be 
+  appended with the underscore (``_``) character. 
+  For matrices (two dimensions) and arrays with more than
+  two dimensions, each index shall be concatenated
+  with the underscore character (``_``).
```

Comment:
You also need to add in normative text how an assignment is serialized. See https://fmi-standard.org/docs/3.0/ section 2.2.7.1 for the order used to serialize a matrix into a one-dimensional string.

Reply:
If the values of an array is being specified in CDL, how will this be specified in CXF? (adhere to FMI)